x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/update_ml_inference_mappings.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+
+import { FieldMapping } from '../../../../../common/ml_inference_pipeline';
+
+import { ErrorCode } from '../../../../../common/types/error_codes';
+
+import { isIllegalArgumentException } from '../../../../utils/identify_exceptions';
+
+/**
+ * Creates Elasticsearch field mappings so that the outputs of ML Inference pipelines are not indexed as `float` fields
+ * @param indexName - the index whose mapping will be updated
+ * @param fieldMappings - the array of objects representing the source field (text) names and target fields (ML output) names
+ * @param esClient
+ */
+export const updateMlInferenceMappings = async (
+  indexName: string,
+  fieldMappings: FieldMapping[],
+  esClient: ElasticsearchClient
+) => {
+  const sourceFields = fieldMappings.map((fieldMapping) => {
+    return fieldMapping.sourceField;
+  });
+
+  const nonDefaultTargetFields = fieldMappings
+    .filter((fieldMapping) => {
+      return fieldMapping.targetField !== `ml.inference.${fieldMapping.sourceField}_expanded`;
+    })
+    .map((fieldMapping) => {
+      return fieldMapping.targetField;
+    });
+
+  // Today, we only update mappings for ELSER-analyzed fields.
+  const mapping = formElserMappingProperties(sourceFields, nonDefaultTargetFields);
+  try {
+    return await esClient.indices.putMapping({
+      index: indexName,
+      properties: mapping,
+    });
+  } catch (e) {
+    if (isIllegalArgumentException(e)) {
+      throw new Error(ErrorCode.MAPPING_UPDATE_FAILED);
+    } else {
+      throw e;
+    }
+  }
+};
+
+const formElserMappingProperties = (sourceFields: string[], targetFields: string[]) => {
```

Comment:
Minor: Ah, after a closer look, now I understand the use of the word "form". 😅  I'd use "generate" instead - I associate "form" with the UI component that takes the input; it might be confusing to others too.

Reply:
Another minor: in the frontend code we tend to refer to ELSER inference as ["text expansion"](https://github.com/search?q=repo%3Aelastic%2Fkibana%20text_expansion&type=code), since that's the name of the ML model task and the condition for such special logic. ELSER is more of a marketing name and it's used on UI labels only.

Reply:
switched to "generate" and "text expansion"